src/Resizer.php

```diff
@@ -111,7 +111,7 @@ protected function executeResize(ImageInterface $image, ResizeCoordinatesInterfa
         }
 
         // Atomic write operation
-        $tmpPath = $this->filesystem->tempnam($dir, 'img');
+        $tmpPath = sys_get_temp_dir().'/'.basename($path);
         $imagineImage->save($tmpPath, $imagineOptions);
```

Comment:
How about using the following code instead?

```php
$imagineImage->save($tmpPath, $imagineOptions + [
    'format' => strtolower(pathinfo($path, PATHINFO_EXTENSION)),
]);
```

Then we could keep the `filesystem->tempnam()` call.